### PR TITLE
Split Javascript white-list to support contrib modules

### PR DESCRIPTION
**Pull request by asmorkalov**
Single whitelist converted to several per-module json files. They are concatenated automatically and can be overriden by user config.

Related to https://github.com/opencv/opencv/pull/25656

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom
Xbuild_image:Docs=docs-js
build_image:Custom=javascript
build_image:Docs=docs-js:22.04-3.1.64
#build_image:Custom=javascript:3.1.64
buildworker:Custom=linux-4
buildworker:Docs=linux-4
```

### CODE REVIEW

File: modules/js/generator/CMakeLists.txt [hunk 1]
```diff
@@ -38,8 +38,23 @@ set(scripts_hdr_parser "${JS_SOURCE_DIR}/../python/src2/hdr_parser.py")
 
 if(DEFINED ENV{OPENCV_JS_WHITELIST})
   set(OPENCV_JS_WHITELIST_FILE "$ENV{OPENCV_JS_WHITELIST}")
+  message(STATUS "Use white list from environment ${OPENCV_JS_WHITELIST_FILE}")
 else()
-  set(OPENCV_JS_WHITELIST_FILE "${OpenCV_SOURCE_DIR}/platforms/js/opencv_js.config.py")
+  #generate white list from modules/<module_name>/misc/js/whitelist.json
+  set(OPENCV_JS_WHITELIST_FILE "${CMAKE_CURRENT_BINARY_DIR}/whitelist.json")
+  set(OPENCV_JS_WHITELIST_CONTENT)
+  file(WRITE "${OPENCV_JS_WHITELIST_FILE}" "{\n")
```
Comment (opencv-alalek) [inline modules/js/generator/CMakeLists.txt:46]: No need to write file each time.
Generate string and update on changes only. See `ocv_update_file`

Also generated file should be added as a dependency to the build graph of bindings generator target.

File: modules/js/generator/embindgen.py [hunk 2]
```diff
@@ -953,8 +955,11 @@ def gen(self, dst_file, src_files, core_bindings):
     headers = open(sys.argv[3], 'r').read().split(';')
     coreBindings = sys.argv[4]
     whiteListFile = sys.argv[5]
-    exec(open(whiteListFile).read())
```
Comment (opencv-alalek) [inline modules/js/generator/embindgen.py:956]: Old behavior seems broken.

Besides of json there are namespaces configuration were possible (see `namespace_prefix_override`), so python configuration code is preferable and provides full customization without unnecessary limitation of json.

**Comment by opencv-alalek** [comment]
Regression in javascript tests (with both legacy and upgraded EMSDK):

```
Errors:

Module: Photo Test: test_photo
Died on test #1: cv.CalibrateDebevec is not a constructor
    at Object.global.test.QUnit.test (/build/precommit_custom_linux/build/bin/node_modules/node-qunit/lib/child.js:132:17)
    at Object.<anonymous> (/build/precommit_custom_linux/build/bin/test_photo.js:45:7)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
TypeError: cv.CalibrateDebevec is not a constructor
    at Object.<anonymous> (/build/precommit_custom_linux/build/bin/test_photo.js:48:27)
```

photo's module part is missing.

**Comment by asmorkalov** [comment]
@opencv-alalek I fixed the namespace issue and added branch with the old python config. Please take a look again.

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**